File: php/src/Advertisements/User/UI/Http/EnableMemberController.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\User\UI\Http;

use Demo\App\Advertisements\User\Application\Command\DisableMember\DisableMemberCommand;
use Demo\App\Advertisements\User\Application\Command\DisableMember\DisableMemberUseCase;
use Demo\App\Advertisements\User\Application\Command\EnableMember\EnableMemberCommand;
use Demo\App\Advertisements\User\Application\Command\EnableMember\EnableMemberUseCase;
use Demo\App\Common\Exceptions\BoundedContextException;
use Demo\App\Common\UI\CommonController;
use Demo\App\Framework\FrameworkRequest;
use Demo\App\Framework\FrameworkResponse;
use Demo\App\Framework\SecurityUser\FrameworkSecurityService;

final class EnableMemberController extends CommonController
{
    public function __construct(
        private EnableMemberUseCase $useCase,
        private FrameworkSecurityService $securityService,
    ) {}

    public function request(FrameworkRequest $request, array $pathValues = []): FrameworkResponse
    {
        $user = $this->securityService->getSecurityUserFromRequest($request);

        if (null == $user || !$user->role() == 'admin') {
            return $this->processUnauthorizedResponse();
        }

        try {
            $command = new EnableMemberCommand(
                $user->id(),
                $user->role(),
                $pathValues['memberId'],
            );

            $this->useCase->execute($command);

            return $this->processSuccessfulCommand();
        } catch (BoundedContextException $exception) {
            return $this->processDomainOrApplicationExceptionResponse($exception);
        } catch (\Throwable $exception) {
            return $this->processGenericException($exception);
        }
    }
}

File: kotlin/src/main/kotlin/advertisements/user/ui/http/EnableMemberController.kt
package advertisements.user.ui.http

import advertisements.user.application.command.enablemember.EnableMemberCommand
import advertisements.user.application.command.enablemember.EnableMemberUseCase
import common.application.ElementNotFoundException
import common.exceptions.BoundedContextException
import common.ui.http.CommonController
import framework.FrameworkRequest
import framework.FrameworkResponse
import framework.securityuser.FrameworkSecurityService

class EnableMemberController(
    private val useCase: EnableMemberUseCase,
    private val securityService: FrameworkSecurityService,
): CommonController() {

    override fun execute(request: FrameworkRequest, pathValues: Map<String, String>): FrameworkResponse {
        try {
            val user = securityService.getSecurityUserFromRequest(request)!!

            if (user.role != "admin") {
                return processUnauthorizedResponse()
            }

            useCase.execute(
                EnableMemberCommand(
                    user.id,
                    user.role,
                    pathValues["memberId"]!!,
                )
            )

            return processSuccessfulCommand()
        } catch (e: ElementNotFoundException) {
            return processNotFoundCommand(e)
        } catch (e: BoundedContextException) {
            return processApplicationOrDomainException(e)
        } catch (e: Exception) {
            return processGenericException(e)
        }
    }
}